### ISSUE TITLE
Getting VOIP to work on hosted servers

### ISSUE THREAD

**Issue by gabrielbunselmeyer** [StunlockStudios/vrising-dedicated-server-instructions#18]
More of a question than an issue (can't get a straight answer on Discord nor on the subreddit): does ingame VOIP work on custom servers, or is it only support on official ones?

Using Google Cloud to host and can't quite figure this one out. I'd assume it works but may require opening other specific ports; no information about it anywhere though.

**Comment by Xect0r** [comment]
> More of a question than an issue (can't get a straight answer on Discord nor on the subreddit): does ingame VOIP work on custom servers, or is it only support on official ones?
> 
> Using Google Cloud to host and can't quite figure this one out. I'd assume it works but may require opening other specific ports; no information about it anywhere though.

They haven't released VOIP for Private Servers, only Official ones are allowed to use VOIP since it's hardcoded into the server. Hopefully, they will allow server owners to choose rather than force them to not have VOIP.

**Comment by paulvilla** [comment]
I would like this, it is necessary that all servers have the VOIP system

**Comment by P4sca1** [comment]
According to the discord, you need to apply for a Vivox API key and then create a `ServerVoipSettings.json` file:
```json
{
    "VOIPEnabled": true,
    "VOIPIssuer": "",
    "VOIPSecret": "",
    "VOIPAppUserId": "",
    "VOIPAppUserPwd": "",
    "VOIPVivoxDomain": "",
    "VOIPAPIEndpoint": "",
    "VOIPConversationalDistance": 14,
    "VOIPAudibleDistance": 40,
    "VOIPFadeIntensity": 2.0
}
```

**Comment by gabrielbunselmeyer** [comment]
Yeah, that seems to be it. I have a Sandbox Vivox account going and so far it works fine! 🎉 
Not sure if we were intended to find this file or not hah, as it's hidden in G-Portal and they were the only host that knew it existed.

**Comment by paulvilla** [comment]
oh yes, as soon as he gets home from work I try it.

PDT: Very ugly on the part of the company that does not give this option when it does not count at all.
Enough that we are creating servers for people to play their game and think that we are undervalued if we do not pay for a server from their sponsor
